CMakeLists.txt

```diff
@@ -249,6 +250,7 @@ option(ENABLE_YANGLINT_INTERACTIVE "Enable interactive CLI yanglint" ON)
 option(ENABLE_TOOLS "Build binary tools 'yanglint' and 'yangre'" ON)
 option(ENABLE_COMMON_TARGETS "Define common custom target names such as 'doc' or 'uninstall', may cause conflicts when using add_subdirectory() to build this project" ON)
 option(BUILD_SHARED_LIBS "By default, shared libs are enabled. Turn off for a static build." ON)
+option(ENABLE_CBOR_SUPPORT "Enable CBOR support with libcbor" ON)
```

Comment:
Actually, an `option` is probably redundant. You can just search for `libcbor` and if found, the support will be added, otherwise not.